src/test/java/de/tum/in/www1/artemis/programmingexercise/ProgrammingExerciseIntegrationTest.java

```diff
@@ -742,6 +742,16 @@ public void createProgrammingExercise_bambooProjectWithSameTitleAlreadyExists_ba
         request.post(ROOT + SETUP, programmingExercise, HttpStatus.BAD_REQUEST);
     }
 
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void createProgrammingExercise_checkoutSolutionRepositoryProgrammingLanguageNotSupported_badRequest() throws Exception {
```

Comment:
can we improve this test using the programming language as enum parameter so that all programming languages except Haskell are tested to produce a bad request? (this would then also include future programming languages

Reply:
I've changed it to a `@ParameterizedTest` that uses all `ProgrammingLanguage`s except `HASKELL`. This is also safe for the future as the test excludes `HASKELL` (as opposed to including all other programming languages).